resproc.c

```diff
@@ -0,0 +1,195 @@
+/* Copyright (C) 2018, Oracle and/or its affiliates. All rights reserved
+ *
+ * This file is part of libresource.
+ *
+ * libresource is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * libresource is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General public License
+ * along with libresource. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef _RESOURCE_H
+#include "resource.h"
+#endif
+#include <stdio.h>
+#include <unistd.h>
+#include <string.h>
+#include "resproc.h"
+#include "resource_impl.h"
+#include <sys/stat.h>
+#include <pwd.h>
+
+/* Reading information from /proc/<pid>/statm */
+static void statm_to_proc(char* buf, res_proc_infoall_t *p) {
+	sscanf(buf, "%ld %ld %ld %ld %ld %ld %ld",
+		&p->size, &p->resident, &p->share,
+		&p->text, &p->lib, &p->data, &p->dt);
+}
+
+/* Reading information from /proc/<pid>/stat */
+static void stat_to_proc(char* buf, res_proc_infoall_t *p) {
+	unsigned int num;
+	char* tmp;
+
+	/* fill in default values for older kernels */
+	p->processor = 0;
+	p->rtprio = -1;
+	p->policy = -1;
+	p->nlwp = 0;
+
+	/* Fill the command to run process */
+	buf = strchr(buf, '(') + 1;
+	tmp = strrchr(buf, ')');
+	num = tmp - buf;
+	if(num >= sizeof p->cmd)
+		num = sizeof p->cmd - 1;
```

Comment:
Hm, are you using `sizeof` without `()` everywhere? For consistency you should probably pick one of the two `sizeof` or `sizeof()`. So far we have always used the latter.